### DISCUSSION TITLE
Error while updating tlmgr...

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#3011] [Q&A] [answered]
I am getting this error:

```
updating tlmgr

updating existing packages
finding package for MinionPro(-(Bold|Italic|Regular).*)?[.](tfm|afm|mf|otf|ttf)
finding package for MinionPro
finding package for mktextfm
> 1 package to install
> installing texlive-scripts (1 of 1)
ERROR: tlmgr returned a non zero status code
You don't have permission to change the installation in any way,
specifically, the directory /usr/local/texlive/2022/tlpkg/ is not writable.
Please run this program as administrator, or contact your local admin.
tlmgr: An error has occurred. See above messages. Exiting.
```
My google searches tell me this is something related to TinyTex (not sure).

If I change permission to `/usr/local/texlive/2022/tlpkg/`  the error changes (now it can't find the font, although I see it in my FontBook).  I am not sure I should have changed the permissions there, though. Nor do I know how to make tlmgr finds a font that is in ...Library/Fonts (mac user).

**Comment by fredguth** [comment]
BTW, I have tex documents that use Minion Font using `\usepackage{MinionPro}` with no problem.

**Reply by fredguth** [reply]
Hmmm... it needs me to call it "Minion Pro" with the space.

Still, don't know if I should have changed the permissions to `/usr/local/texlive/2022/tlpkg/`, though.  At least it is working.

**Comment by mcanouil** [comment, marked as answer]
What version of Quarto are you using? What's the output of `quarto list tools`?
v1.2 pre-release has a bunch of changes related to Tex.
See <https://quarto.org/docs/output-formats/pdf-engine.html#installing-tex>.

By the way, with only the error, it's really hard to help. It would be really helpful if you could provide some context, such as operating system, Quarto version, LaTeX distribution, a reproducible example.

**Reply by fredguth** [reply]
```
❯ quarto list tools
[✓] Inspecting tools

Tool         Status            Installed     Latest
chromium     Not installed     ---           869685
tinytex      Not installed     ---           v2022.10

````

**Reply by mcanouil** [reply]
What version of Quarto?

Since you are apparently not using Quarto TinyTex installation. I would suggest to use the pre-release version and set `latex-tinytex: false` to use your other Tex distribution.
Or to use the new local installation of TinyTex by Quarto to have a clean tex distribution.
All that is described in <https://quarto.org/docs/output-formats/pdf-engine.html>.

If you have some issues with this and/or the documentation is not clear enough, please report back so that Quarto team can try to improve it. 😉

**Reply by fredguth** [reply]
```
❯ quarto --version
1.2.215
```
> Since you are apparently not using Quarto TinyTex installation. I would suggest to use the pre-release version and set latex-tinytex: false to use your other Tex distribution.

how to set latex-tinytex: false?  shouldn't I use the TinyTex distribution?  I don't understand much about Tex distributions.  I though I was using tinyTex automatically by just using Quarto.

**Reply by mcanouil** [reply]
Oh, my bad, wrong assumption. 🤦‍♂️

Quarto uses some tools internally but it can use what's already available on the computer without requiring to install anything.
This is done, to allow the user to have Quarto adapts to them, not the other way around.

For convenience, Quarto offers to the user a nice and internal way to have access to LaTeX in our case which does not require much knowledge to use it.
TinyTex is a lightweight LaTeX distribution which is a collection of LaTeX libraries/packages with some nice features allowing it to only install what is needed when you compile tex files to PDF, either yourself or via a tool such as Pandoc (used by RMarkdown and Quarto).
Tinytex is maintained by RStudio/Posit, which is one of the reason (I suppose) it was integrated into Quarto.
Because Quarto does not assume you are going to need LaTeX, it does not install anything, thus you have to do it yourself.

Hope this clarify things for you.

**Reply by mcanouil** [reply]
Now before actually following the documentation to install Tinytex, there is one question to address: do you compile tex files to PDF outside of Quarto?

If the answer is no, then first:
- upgrade Quarto to the latest pre-release version: <https://quarto.org/docs/download/>.
- use `quarto install tinytex` (see details in <https://quarto.org/docs/output-formats/pdf-engine.html#installing-tex>).
  This command will install Tinytex only for Quarto, which will avoid conflict with any LaTeX distribution already installed on your computer.
  It will also or at least should make it easier for you to use Quarto and PDF-based formats.

If the answer is yes:

- set the following yaml option (in your yaml header or in `_quarto.yml`): `latex-tinytex: false`.
  This will tell Quarto to not try to use Tinytex but instead to try to use whatever LaTeX distribution available.  
  This also means, you'll have to deal yourself with missing LaTeX packages and install those yourself. (This is mainly why, Quarto recommends the use of Tinytex).

**Reply by fredguth** [reply]
I used to compile outside of Quarto, but I want to migrate everything to Quarto. So, I understand that if I go with the "no" (first) option, I will be able to eventually use the distribution I already have (I believe it is TexLive), *but*  I will have a better Quarto experience, right?

I am inclined to the first option.

**Reply by mcanouil** [reply]
Yes.
Note that with the first option, you will use two different LaTeX distribution: one for Quarto and one for everything else.

**Reply by fredguth** [reply]
:+1:

**Reply by mcanouil** [reply]
@fredguth I assumed that you solved your issue by using Quarto TinyTex? (Just checking to be sure your issue is solved since you marked this as the answer)

**Reply by fredguth** [reply]
Yes, it seems solved. My impressions:
- it seems that compiling is faster. Can it be? 
- tinytex didn't accept:
```_extentension.yml
            mainfontoptions: [minionint,footnotefigures, frenchmath] # <--- mainfont is Minion Pro
```
but this is a minor issue for me.

```
❯ quarto list tools; quarto --version
[✓] Inspecting tools

Tool         Status            Installed     Latest
chromium     Not installed     ---           869685
tinytex      Up to date        v2022.10      v2022.10
1.2.242
```
